### ISSUE TITLE
Ampersand & inside pseudo selectors :not() not resolved anymore

### ISSUE THREAD

**Issue by troch** [postcss/postcss-nested#64]
Hi,

Use of `&` within pseudo selectors like `:not` is broken on v2.  The following example:

```css
.a {
	&:not(&--b) {
		display: none;
	}
}
```

On version 1.x, it was transformed to:

```css
.a:not(.a--b) {
    display: none;
}
```

On version 2.x, it is transformed to:

```css
.a:not(&--b) {
    display: none;
}
```

**Comment by ai** [comment]
Yeap, it is a bug. Thanks for the report. I will try to look at in tomorrow.

**Comment by troch** [comment]
Thanks!

**Comment by ai** [comment]
Done 1921ceb

**Comment by ai** [comment]
Released in 2.1.1

**Comment by troch** [comment]
Thanks a lot for the quick fix!

**Comment by troch** [comment]
@ai it appears the fix only works when there is one`&` in the rule.

The following works:

```css
.component {
   :not(&--blue) {
      color: blue;
   }
}
```

But the following doesn't work

```css
.component {
   &--highlighted:not(&--blue) {
      color: blue;
   }
}
```

**Comment by ai** [comment]
@troch will try to look in next days.

**Comment by ai** [comment]
Fixed bfc0b907b39b0114d2a32748097ed04a2f842592

**Comment by ai** [comment]
Released in 2.1.2

**Comment by troch** [comment]
All good, confirmed as fixed 👍

Thanks!
